src/main/java/uk/gov/hmcts/reform/refunds/services/IdamServiceImpl.java

```diff
@@ -105,27 +111,62 @@ public UserIdentityDataDto getUserIdentityData(MultiValueMap<String, String> hea
                 .build();
         }
 
-        ResponseEntity<IdamFullNameRetrivalResponse[]> idamFullNameResEntity = restTemplateIdam
+        ResponseEntity<IdamUserInfoResponse[]> idamFullNameResEntity = restTemplateIdam
             .exchange(
                 builder.toUriString(),
                 HttpMethod.GET,
-                getEntity(headers), IdamFullNameRetrivalResponse[].class
+                getEntity(headers), IdamUserInfoResponse[].class
             );
 
-
         if (idamFullNameResEntity != null && idamFullNameResEntity.getBody() != null) {
-            IdamFullNameRetrivalResponse[] idamArrayFullNameRetrievalResponse = idamFullNameResEntity.getBody();
+            IdamUserInfoResponse[] idamArrayFullNameRetrievalResponse = idamFullNameResEntity.getBody();
 
             if (idamArrayFullNameRetrievalResponse != null && idamArrayFullNameRetrievalResponse.length > 0) {
-                IdamFullNameRetrivalResponse idamFullNameRetrivalResponse = idamArrayFullNameRetrievalResponse[0];
+                IdamUserInfoResponse idamUserInfoResponse = idamArrayFullNameRetrievalResponse[0];
                 return UserIdentityDataDto.userIdentityDataWith()
-                    .emailId(idamFullNameRetrivalResponse.getEmail())
-                    .fullName(idamFullNameRetrivalResponse.getForename() + " " + idamFullNameRetrivalResponse.getSurname())
+                    .emailId(idamUserInfoResponse.getEmail())
+                    .fullName(idamUserInfoResponse.getForename() + " " + idamUserInfoResponse.getSurname())
                     .build();
             }
         }
 
         LOG.error("User name not found for given user id : {}", uid);
-        throw new UserNotFoundException("Internal Server error. Please, try again later");
+        throw new UserNotFoundException(INTERNAL_SERVER_ERROR);
+    }
+
+    @Override
+    public Set<String> getUserIdSetForService(MultiValueMap<String, String> headers, List<String> roles) {
+        /*Set<String> users = new HashSet<>();
+        users.add("asdfghjk-kjhgfds-dfghj-sdfghjk");
+        return users;*/
+
+//        Pattern rolePattern = Pattern.compile("^.*damage.*$");
+
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(idamBaseURL + USER_FULL_NAME_ENDPOINT)
+                .queryParam("query", "(roles:" + roles + ") AND lastModified:" + LAST_MODIFIED_TIME)
```

Comment:
add >now- to the lastmodified in the query

Reply:
Done